lightstep/recorder.py

```diff
@@ -223,15 +276,48 @@ def _add_span(self, span):
 
         Will delete a previously-added span if the limit has been reached.
         """
+        now_micros = util._now_micros()
+        
+        spanRecord = ttypes.SpanRecord(
+            trace_guid=str(span.context.trace_id),
+            span_guid=str(span.context.span_id),
+            runtime_guid=str(span._tracer.recorder._runtime_guid),
+            span_name=str(span.operation_name),
+            join_ids=[],
+            oldest_micros=long(span.start_time),
+            youngest_micros = long(now_micros),
+            attributes=[],
+            log_records=[]
+        )
+
+        for key in span.tags:
+            if key[:5] == "join:":
+                spanRecord.join_ids.append(ttypes.TraceJoinId(key, span.tags[key]))
+            else:
+                spanRecord.attributes.append(ttypes.KeyValue(key, span.tags[key]))
+
+        for log in span.logs:
+            event = ""
+            if len(log.event)>0:
+                #Don't allow for arbitrarily long log messages.
+                if sys.getsizeof(log.event)>constants.MAX_LOG_MEMORY:
+                    event = log.event[:constants.MAX_LOG_LEN]
+                else:
+                    event = log.event
+            spanRecord.log_records.append(ttypes.LogRecord(stable_name= event, payload_json= str(log.payload)))
```

Comment:
I think I see what you mean. I converted it there because somewhere down the line, deep in thrift cod, it expects a string-- specifically the error is 'must be string or buffer, not dict' occurring in `thrift/transport/THttpClient.py`. Not sure how to best address that issue 😕